node/npm_dependencies.ts

```diff
@@ -62,34 +134,48 @@ export const vendorNPMSpecifiers = async (basePath: string, functions: string[],
   // resolution logic won't work.
   const nodePaths = [path.join(basePath, 'node_modules')]
 
+  // We need to create some files on disk, which we don't want to write to the
+  // project directory. If a custom directory has been specified, which happens
+  // only in tests, we use it. Otherwise, create a random temporary directory.
+  const temporaryDirectory = directory ? { path: directory } : await tmp.dir()
```

Comment:
Moved this up so that we can pass it as `outdir` to the first `build` call. That shouldn't be needed, because we're setting `write: false`, so we won't write any files to disk anyway. But esbuild still complains, which I think is a bug.